wordpress-mobile/WordPress-FluxC-Android
# ScanStore: Fetch Scan State

Issue: wordpress-mobile/WordPress-Android#13326

Jetpack Scan APIs [Internal Ref `pcdRpT-3J`]

`/sites/$site/scan` API is used to
- know if the scan feature is unlocked for a site
- get progress of a current scan
- get a list of threats for a site.

This PR introduces `/sites/$site/scan` API to the `ScanStore` and adds corresponding model classes, `ScanAction`, fetch methods in the `ScanRestClient` and `ScanStore`.

#### To test

- Add a breakpoint on this line 
https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/90f65e413066434adfcda59552a8cc12ee93778a/fluxc/src/main/java/org/wordpress/android/fluxc/store/ScanStore.kt#L54

- debug temporary test `testFetchScanState` in the `ReleaseStack_ScanTestJetpack` to see fetch results in the `payload.scanStateModel` at the breakpoint above.

You might need to update `tests.properties` if you haven't already to get jetpack complete site details.

#### Note
API response is not persisted anywhere yet. Storing the data to DB and getting from it via `ScanStore` will be supported in an upcoming PR. Tests will also be added at that point.


#### Questions
1. As we use the same API for different actions listed above, I'm in dilemma if there should be separate actions to 

   - get just the "scan feature unlocked status" [with a separate model to just contain information about this status]
   - get progress of a current scan
   - get list of threats for a site. 


    Let me know if you have suggestions.

2. ~Is it fine to map both `most_recent` and `current` to a common object `ScanProgressStatus` (See c54b4e4)? Seems like only one of them appear in the response based on the scan state and they have common fields `is_initial`, `timestamp` and `progress`.~ I've kept them separate in 799a7b4

#### Todo
- ~Map "current" object when state is "scanning"~ Done in c54b4e4
- ~Map "reason"~ Done in 5a7e6b9
